react/hooks/mapCustomTypes.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import EnumTable from '../components/EnumTable'
+import ObjectsTable from '../components/ObjectsTable'
+import ArrayTable from '../components/ArrayTable'
+
+export const mapCustomTypes = (
+  propsObj: Record<string, ObjSchemaInterface>,
+  messages: Record<string, Record<string, string>>
+) => {
+  let mappedTypes: JSX.Element[] = []
+  let mappedProps: Record<string, ObjSchemaInterface> = propsObj
+  for (const key in mappedProps) {
+    let currentComponent: JSX.Element | null = null
+    const currentProp = mappedProps[key]
+    if (currentProp.enum) {
+      currentProp.type = `${key.charAt(0).toLocaleUpperCase()}${key.slice(
```

Comment:
wrap this logic in function to improve readability

Reply:
which logic part?